src/app/AttributeAccessInterface.h

```diff
@@ -414,6 +414,34 @@ class AttributeAccessInterface
      */
     virtual CHIP_ERROR Write(const ConcreteDataAttributePath & aPath, AttributeValueDecoder & aDecoder) { return CHIP_NO_ERROR; }
 
+    /**
+     * Indicates the start of a series of list operations. This function will be called before the first Write operation of a series
+     * of consequence attribute data of the same attribute.
+     *
+     * 1) This function won't be called if the user writes a null value to a list attribute.
+     * 2) This function will only be called once for a series of consequent attribute data (regardless the kind of list operation)
+     * of the same attribute.
+     *
+     * @param [in] aPath indicates the path of the modified list.
+     */
+    virtual void OnListWriteBegin(const ConcreteAttributePath & aPath) {}
+
+    /**
+     * Indicates the end of a series of list operations. This function will be called after the last Write operation of a series
+     * of consequence attribute data of the same attribute.
+     *
+     * 1) This function won't be called if the user writes a null value to a list attribute.
+     * 2) This function will only be called once for a series of consequent attribute data (regardless the kind of list operation)
+     * of the same attribute.
+     * 3) When aWriteWasSuccessful is true, the data written must be consistent or the list is untouched.
+     * 4) When aWriteWasSuccessful is false, the list might be untouched.
```

Comment:
I'm not sure what 'untouched' means? It certain might have been touched. The point here is that the list transaction did not complete successfully, which means any mutations to the list since the `OnListWriteBegin` should be deleted if possible.

Reply:
The untouched means the attribute accessing interface might receive nothing between begin and end (the data may failed to pass the access control checks)

Seems this line is redundant since the attribute access interface is expected to revert the content anyway.